controllers/openshift_controller.go

```diff
@@ -66,14 +64,14 @@ type KataConfigOpenShiftReconciler struct {
 	kataConfig         *kataconfigurationv1.KataConfig
 	FeatureGates       *featuregates.FeatureGates
 	FeatureGatesStatus featuregates.FeatureGateStatus
+	deploymentState    DeploymentState
```

Comment:
I'm quite afraid this breaks the controller's idempotency.  The controller can be migrated to another worker at any time and if I'm not mistaken this will wipe the deployment state information.  This is not a theoretical concern either.  The controller normally runs on workers so any worker reboot is an opportunity for the controller to be killed and restarted elsewhere.  I've seen many times during kata deployment on all workers of a two-worker cluster the controller being migrated twice (ie. the number of workers).  This happens when the controller happens to be running on the first worker to get kata deployed.  It's then migrated to the second worker, only to be migrated back to the first worker when the second worker has kata deployed on it subsequently.

Instead of storing the deployment state here, we should ideally figure out the deployment state by examining the cluster.  In case that's impossible or impractical (I don't actually know if that's possible but I suspect it might be) the state should still be stored somewhere in the cluster, perhaps in `KataConfig.status`.